pkg/ring/consul_client.go

```diff
@@ -245,6 +298,11 @@ func (c *prefixedConsulClient) WatchKey(ctx context.Context, key string, f func(
 	c.consul.WatchKey(ctx, c.prefix+key, f)
 }
 
+// WatchPrefix watches a prefix. For a prefix client it appends the prefix argument to the clients prefix.
+func (c *prefixedConsulClient) WatchPrefix(ctx context.Context, prefix string, f func(interface{}) bool) {
+	c.consul.WatchPrefix(ctx, fmt.Sprintf("%s/%s", c.prefix, prefix), f)
```

Comment:
Once `f` receives a path, it would make sense for the prefix to be removed here.

Reply:
Ping?

Reply:
```go
c.consul.WatchPrefix(ctx, fmt.Sprintf("%s/%s", c.prefix, prefix), func(k string, i interface{}) bool {
  return f(strings.TrimPrefix(key, c,prefix), i)
})
```